warehouse/query-core/src/main/java/datawave/query/function/IndexOnlyContextCreator.java

```diff
@@ -190,7 +185,7 @@ protected DatawaveJexlContext newDatawaveJexlContext(final Tuple3<Key,Document,M
         }
         
         // see if there are any delayed nodes that need to be processed
-        if (delayedNonEventFieldMap != null && delayedNonEventFieldMap.keySet().size() > 0) {
+        if (delayedNonEventFieldMap != null && !delayedNonEventFieldMap.keySet().isEmpty()) {
```

Comment:
I think delayedNonEventFieldMap.isEmpty() is what you want here.  This avoids the creation of a key set and achieves the same purpose.

Reply:
ok